### DISCUSSION TITLE
Github Pages are not published after following the documentation

### DISCUSSION THREAD

**Discussion by dotnetjunkie** [community/community#22506] [Other Feature Feedback, Questions, & Ideas]
<p>I know this question has been asked several times before, but none of the answers given seem to fix my problem. I followed the documentation (or at least, I believe I did, as I read through it several times), but I seem to be missing something:</p>
<p>What have I got:</p>
<ul>
<li>The account: <a href="https://github.com/simpleinjector/" rel="nofollow noopener">https://github.com/simpleinjector/</a>
</li>
<li>The public repository: <a href="https://github.com/simpleinjector/simpleinjector.github.io" rel="nofollow noopener">https://github.com/simpleinjector/simpleinjector.github.io</a>
</li>
<li>A <em>master</em> branch in that <em><a href="http://simpleinjector.github.io" rel="nofollow noopener">simpleinjector.github.io</a></em> repository</li>
<li>The GitHub Pages settings of the <em><a href="http://simpleinjector.github.io" rel="nofollow noopener">simpleinjector.github.io</a></em> repository is set to the <em>master</em> branch, and state that “Your GitHub Pages site is currently being built from the master branch.”</li>
<li>I last pushed html files to <em><a href="http://simpleinjector.github.io" rel="nofollow noopener">simpleinjector.github.io</a></em> 18 hours ago.</li>
</ul>
<p>This is my current GitHub Pages settings tab:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/ee6af0d84b88bca1f2b0340aac7bf9ecc104601f.png" data-download-href="/uploads/short-url/y18KY7lp53t3lZHFg5VaiOahd6v.png?dl=1" title="Github pages settings simpleinjector.github.io.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/1506621/181071535-4e71c7d6-887b-4863-916c-0332f7ad8352.png" alt="Github pages settings simpleinjector.github.io.png" data-base62-sha1="y18KY7lp53t3lZHFg5VaiOahd6v" width="690" height="321" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/ee6af0d84b88bca1f2b0340aac7bf9ecc104601f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Github pages settings simpleinjector.github.io.png</span><span class="informations">773×360 15.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>What I haven’t got:</p>
<ul>
<li>The GitHub Page <a href="https://simpleinjector.github.io/" rel="nofollow noopener">https://simpleinjector.github.io/</a> keeps giving a 404, even after 20 hours of first pushing files to the repository.</li>
<li>The GitHub Page settings does not state that the build failed, nor that it succeeded. I’ve seen screenshots on Stack Overflow that show that the settings tab should show either a success or error message. This indicates to me that for some reason Github has no idea that this repository should be the GitHub Page repo.</li>
</ul>
<p>There must be something I am forgetting, but I feel that I followed the documentation to the letter. Can anyone point out my mistake?</p>

**Comment by dotnetjunkie** [comment]
<p>Absolutely. The following link does not serve the index.html as you would expect:</p>
<ul>
<li><a href="https://simpleinjector.github.io/index.html" rel="nofollow noopener">https://simpleinjector.github.io/index.html</a></li>
</ul>
<p>It think this proves the repository is not published on <a href="http://github.io" rel="nofollow noopener">github.io</a>.</p>

**Comment by dotnetjunkie** [comment, marked as answer]
<p>In order for your site to be published, a Jekyll theme must be chosen, even if you don’t use Jekyll <em>at all</em>.</p>

**Reply by thihamz** [reply]
I do I choose a Jekyll theme?

**Reply by hehewhoknows** [reply]
thanks alot it worked 🥳
{
make a _config.yml in the branch from which youre deploying the site and then write 'theme: minima' {without the quotes} feel free to replace themes!  
after comitting the file should be deployed almost instantly
}

**Reply by jdocceno** [reply]
11/29/2023, thanks a lot hehewhoknows I followed your instructions and it helped me display my site live on github.io

**Reply by caijianhong** [reply]
Thank you!

**Reply by SurajGaur97** [reply]
@hehewhoknows Thankyou, it worked for me and instantly I can see the project deployed.

**Reply by nicolasshu** [reply]
This worked for me too. Thank you!

**Reply by Natnaeldefaru12** [reply]
I didn't think it would work, but it worked. For clarification because I was confused: In the branch you pushed your website, create a file and name it _config.yml then inside it type "theme: minima" without the quotation, commit the changes, then go to settings, then pages, to find your uploaded page.

**Comment by ghost** [comment]
<p>SOLVED</p><blockquote><hr></blockquote>

**Comment by laurenwestonn** [comment]
<p>WOW this actually made my site show up. THANK YOU.</p>
<p>I’ve been trying all sorts to get my site to show. Who knew you had to set a theme?! How does that make any sense?! I’ve not read that anywhere else</p>

**Comment by Tom-Okafor** [comment]
<p>please how long did it take for the site to show up after selecting the theme</p>

**Comment by LBBO** [comment]
<p>For me, it appeared instantly.</p>

**Comment by mnlyz** [comment]
<p>IDK HOW that made it word, but it did. Assigning a theme made everything publish - thank you!</p>

**Comment by sbrahmasamudra** [comment]
<p>How did you attach the theme guys?</p>

**Comment by rajkamal8** [comment]
<p>do you have answer know?if yes, could you tell me please</p>

**Comment by rajkamal8** [comment]
<p>what you did could you explain ?</p>

**Comment by euanmillar** [comment]
<p>The answer is to add Jekyll theme. You do this on the settings tab of the repository of your github pages site.  The official documentation from Github on this page <a href="https://pages.github.com/" rel="nofollow noopener">https://pages.github.com/</a> is missing this vital step.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/c74a43644fbeb12bdb7b6f44aea2af7cd403adcd.png" data-download-href="/uploads/short-url/sr07K6xoYdnRuteUygzzqkv3CbP.png?dl=1" title="Screenshot 2020-04-27 at 09.18.17.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/3169954/181071541-f7d92690-1221-4c6b-bc9c-7f63dca6069f.png" alt="Screenshot 2020-04-27 at 09.18.17.png" data-base62-sha1="sr07K6xoYdnRuteUygzzqkv3CbP" width="690" height="126" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/c/c74a43644fbeb12bdb7b6f44aea2af7cd403adcd_2_690x126.png, https://aws1.discourse-cdn.com/github/original/2X/c/c74a43644fbeb12bdb7b6f44aea2af7cd403adcd.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/c/c74a43644fbeb12bdb7b6f44aea2af7cd403adcd.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/c74a43644fbeb12bdb7b6f44aea2af7cd403adcd_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-04-27 at 09.18.17.png</span><span class="informations">755×138 11 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by Guleed123** [comment]
<p>I did this but still doesnt work </p>

**Comment by Prinsn** [comment]
<p>Also did this, no change, still seing readme</p>

**Comment by poornimashrivastava** [comment]
<p>Hey! I added the theme but still no progress. Can you please elaborate how what to do after changing the theme? Like do I need to do something with the markdown files or any other file?</p>
<p>Thanks</p>

**Comment by poornimashrivastava** [comment]
<p>Hey! I added the theme but still no progress. Can you please elaborate how what to do after changing the theme? Like do I need to do something with the markdown files or any other file?</p>

**Comment by cheryltky** [comment]
<p>I added the jekyll theme as well but still had the same issue with just the readme.md being published only. Anyone have any updates as to what they did differently to get the website running?</p>

**Comment by MaryAllie2006** [comment]
<p>Once you have added a theme you have to add index.html like I did for this project, <a href="https://maryallie2006.github.io/TowerSiege/index.html" rel="noopener nofollow ugc">https://maryallie2006.github.io/TowerSiege/index.html</a></p>

**Comment by therealjgoin** [comment]
<p>This does not work for Hugo sites, by the way.  Still broken.</p>

**Comment by rahulrajpl** [comment]
I unpublished the site. Then did the following steps to solve my issue for Hugo Site.

1. Goto Actions Tab
2. Open Pages build and Deployment
3. Press 're-runn all'

That's it.

**Comment by Callylove** [comment]
Please am currently runing through some problem during the deployment process. it shows deployment successful but only to visit the site and its showing me my README file Here is the live site https://callylove.github.io/New_Year_Countdown.

**Comment by mthtclone** [comment]
I'm late to the party, but the fact that you have to add a Jekyll theme to deploy your site, even if you are not going to use or care about it, doesn't make sense to me.
